packages/react-router/lib/rsc/server.ssr.tsx

```diff
@@ -8,6 +8,21 @@ import { RSCRouterGlobalErrorBoundary } from "./errorBoundaries";
 import { shouldHydrateRouteLoader } from "../dom/ssr/routes";
 import type { RSCPayload } from "./server.rsc";
 
+const react18compatibleUse = (() => {
+  type Usable<T> = Promise<T> | React.Context<T>;
+
+  const errorUse = function <T>(_: Usable<T>): T {
+    throw new Error("React Router v7 requires React 19+ for RSC features.");
+  };
+
+  const functionName = "use";
+  const realUse = (React as any)[functionName];
+
+  return function <T>(promise: Usable<T>): T {
+    return (realUse || errorUse)(promise);
+  };
+})();
```

Comment:
Minor adjustment to use the pattern we've followed for this type of stuff before: https://github.com/remix-run/react-router/blob/v6/packages/react-router-dom/index.tsx#L410

```suggestion
// Safe version of React.use() that will not cause compilation errors against
// React 18 and will result in a runtime error if used (you can't use RSC against
// React 18).
const REACT_USE = "use";
const useImpl = (React as any)[REACT_USE];

function useSafe<T>(promise: Promise<T> | React.Context<T>): T {
  if (useImpl) {
    return useImpl(promise);
  }
  throw new Error("React Router v7 requires React 19+ for RSC features.");
}
```